src/wallet/rpcwallet.cpp

```diff
@@ -2936,7 +2939,12 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
     if (addrType == ADDR_TYPE_SPROUT) {
         return keyIO.EncodePaymentAddress(pwalletMain->GenerateNewSproutZKey());
     } else if (addrType == ADDR_TYPE_SAPLING) {
-        return keyIO.EncodePaymentAddress(pwalletMain->GenerateNewSaplingZKey());
+        auto saplingAddress = pwalletMain->GenerateNewLegacySaplingZKey();
+        if (saplingAddress.has_value()) {
+            return keyIO.EncodePaymentAddress(saplingAddress.value());
+        } else {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "No legacy HD seed available; please use z_getunifiedaddress instead.");
```

Comment:
This is no longer the case; z_getnewaddress now uses the mnemonic seed as the source of randomness. We can still disable it for new wallets if we choose, but that should probably involve a deprecation cycle to avoid disruption to the parts of the ecosystem that may not yet be prepared to use Orchard.